pco/src/data_types/floats.rs

```diff
@@ -147,7 +150,121 @@ macro_rules! impl_float_number {
         l as Self
       }
     }
+  };
+}
+
+impl FloatLike for f16 {
+  const BITS: Bitlen = 16;
+  const PRECISION_BITS: Bitlen = Self::MANTISSA_DIGITS as Bitlen - 1;
+  const ZERO: Self = f16::ZERO;
+  const MAX_FOR_SAMPLING: Self = f16::from_bits(30719); // Half of MAX size.
+
+  #[inline]
+  fn abs(self) -> Self {
```

Comment:
Perhaps we should instead use:
``` rust
    fn abs(self) -> Self {
        Self::from_bits(self.to_bits() & 0x7FFF)
    }
```
Seems like it should be faster. I'm a little bit confused as to why `half` has two different implementations of `abs`:
https://docs.rs/half/latest/src/half/num_traits.rs.html#221
https://docs.rs/half/latest/src/half/num_traits.rs.html#386

Perhaps it would be even better to simply use the `abs` implementation from `half`. However, I couldn't figure out how to do that.

Reply:
I agree this should be faster. I think the code you pasted should work? What error are you getting?

Reply:
Perhaps a better solutions would be to rely on `num_traits` which provides implementations for `abs`, `inv` (named `recip`), `round`, `exp2`, `from_f64`, `to_f64`, `max`, and `min`. What do you think?

Reply:
Not getting any errors with the code i pasted. Pushed the change now. The issue was when i tried calling `self.abs()`. I suppose it's because we don't have the `num_traits`. For now we'll just use our own `Self::from_bits(self.to_bits() & 0x7FFF)` implementation.